src/ConfigEditor/migrating-from-datasource-http-settings.md

```diff
@@ -0,0 +1,88 @@
+
+As `DataSourceHttpSettings` is considered deprecated, `@grafana/experimental` package has a few components that will help you to migrate away from it.
+
+According to the [datasource configuration best practices](https://docs.google.com/document/d/11XMaYHSMSra8AoN2-yVW-5QPbjEXgkAFbTIF6N84cGE/edit) the data source configuration page should consist of multiple sections, so there is no single component that can just replace `DataSourceHttpSettings` component. Instead, there are 3 components that will take care of different parts of the `DataSourceHttpSettings` and they can be placed in different places inside the datasource config editor. This is how mapping old -> new looks like:
+
+<img src="./docs-img/migrate-from-datasource-http-settings.png" width="600">
+
+New components are:
+
+- [ConnnectionSettings](./Connection)
+- [Auth](./Auth)
+- [AdvancedHttpSettings](./AdvancedSettings)
+
+Each of the component has it's own README file and some components (like `Auth`) are highly configurable, so feel free to explore the docs.
+
+## Usage example
+
+This is how the simple migration would look like. You need to remove the old component:
+
+```tsx
+<DataSourceHttpSettings
+  dataSourceConfig={props.options}
+  onChange={props.onOptionsChange}
+  // ... other props
+/>
+```
+
+The new components usage looks like this:
+
+```tsx
+import {
+  ConfigSection,
+  ConnectionSettings,
+  Auth
+  AdvancedHttpSettings,
+  convertLegacyAuthProps,
+} from '@grafana/experimental'
+
+export const ConfigEditor = (props: Props) => {
+  // ...
+
+  return (
+    {/* ... */}
+
+      <ConnectionSettings
+        config={props.options}
+        onChange={props.onOptionsChange}
+      />
+
+      <Auth
+        {...convertLegacyAuthProps({
+          config: props.options,
+          onChange: props.onOptionsChange,
+        })}
+      />
+
+      {/* Other sections if applicable */}
+
+      <ConfigSection
+        title="Advanced settings"
+        isCollapsible
+        isInitiallyOpen={/* if any of the advanced settings is enabled */}
+      >
+        {/* Other sub sections if applicable */}
+
+        <AdvancedHttpSettings
+          config={props.options}
+          onChange={props.onOptionsChange}
+        />
+
+        {/* Other sub sections if applicable */}
+      </ConfigSection>
+
+    {/* ... */}
+  )
+}
+```
+
+> ❗️ Note: As data sources might have different advanced options, a decision was made to provide `AdvancedHttpSettings` as a sub section and not as a whole section, so that other advanced settings can be added alongside advanced HTTP settings. This means that you need to create "Advanced settings" section that wraps `AdvancedHttpSettings` yourself (like in the example above).
+
+## More complex migrations
+
+In case you need to add more fields inside connection section or inside advanced HTTP settings you will need to implement those sections yourself. Feel free to copy the code from those components and extend it with the fields you need.
```

Comment:
AdvancedHttpSettings instead of "advanced HTTP settings"?

Reply:
Done in a3fcb9bd1b9aaa410b4d4f830e16f53db89da98a.